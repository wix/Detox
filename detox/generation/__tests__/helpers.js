const { methodNameToSnakeCase } = require('../helpers');

describe('helpers', () => {
  describe('methodNameToSnakeCase', () => {
    it('should not fail with empty string', () => {
      expect(() => methodNameToSnakeCase('')).not.toThrow();
    });

    it('should return the correct snake case method name', () => {
      expect(methodNameToSnakeCase('actionForScrollInDirection:amount:xOriginStartPercentage:yOriginStartPercentage:')).toMatchSnapshot();
    });
  });
});
