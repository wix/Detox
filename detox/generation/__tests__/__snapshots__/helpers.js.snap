// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`helpers methodNameToSnakeCase should return the correct snake case method name 1`] = `"actionForScrollInDirectionAmountXOriginStartPercentageYOriginStartPercentage"`;
